### ISSUE TITLE
Have an operation on one entity return another

### ISSUE THREAD

**Issue by hcsaustrup** [mgonto/restangular#866]
Hi there!

First of all, thank you for Restangular - it's awesome :-)

I'm building an ecommerce solution, and I've got an API for shopping cart operations like adding, updating or removing an item. I use operations like this:

// Get the card
Restangular.one('cart')

// Add an item to the cart
Restangular.one('cart').one('item').customPOST({ item data })

// Delete an item from the cart
Restangular.one('cart').delete('item')

Now, I've got this idea that any request on the cart or cart items should always return a new instance of the cart, saving a separate request after modifying it. The reason I'm not just manually removing or adding items on the frontend, is that I believe the backend should have the final say-so in product price calculation etc., and thus a new cart should be generated and returned.

So, short version - can I make Restangular return something other than the requested entity, or am I trying to do something completely insane? :-)

Cheers,
HC

**Comment by daviesgeek** [comment]
I think what you're saying is that for every request, you want the entire object to be returned, so that you can always have up-to-date information from the backend?

**Comment by hcsaustrup** [comment]
You git it :-)
